src/test/regress/expected/failure_1pc_copy_hash.out

```diff
@@ -0,0 +1,406 @@
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+TRUNCATE copy_test;
+UPDATE pg_dist_shard_placement SET shardstate = 3 WHERE nodeport = :worker_1_port;
+SELECT * FROM pg_dist_shard_placement WHERE shardid IN (
+  SELECT shardid FROM pg_dist_shard WHERE logicalrelid = 'copy_test'::regclass
+);
+ shardid | shardstate | shardlength | nodename  | nodeport | placementid 
+---------+------------+-------------+-----------+----------+-------------
+  100400 |          3 |           0 | localhost |    57637 |         100
+  100400 |          1 |           0 | localhost |    57640 |         101
+(2 rows)
+
+-- ==== okay, run some tests where there's only one active shard ====
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9' WITH CSV;
+SELECT * FROM copy_test;
+ key | value 
+-----+-------
+   0 |     0
+   1 |     1
+   2 |     4
+   3 |     9
+(4 rows)
+
+-- the worker is unreachable
+SELECT citus.mitmproxy('conn.killall()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9' WITH CSV;
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:57640
+COPY copy_test, line 1: "0, 0"
+ERROR:  failure on connection marked as essential: localhost:57640
+CONTEXT:  COPY copy_test, line 1: "0, 0"
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT * FROM copy_test;
+ key | value 
+-----+-------
+   0 |     0
+   1 |     1
+   2 |     4
+   3 |     9
+(4 rows)
+
+-- the first message fails
+SELECT citus.mitmproxy('conn.onQuery(query="assign_distributed_transaction_id").killall()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9' WITH CSV;
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:57640
+COPY copy_test, line 1: "0, 0"
+ERROR:  failure on connection marked as essential: localhost:57640
+CONTEXT:  COPY copy_test, line 1: "0, 0"
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT * FROM copy_test;
+ key | value 
+-----+-------
+   0 |     0
+   1 |     1
+   2 |     4
+   3 |     9
+(4 rows)
+
+-- the COPY message fails
+SELECT citus.mitmproxy('conn.onQuery(query="FROM STDIN WITH").killall()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9' WITH CSV;
+ERROR:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+CONTEXT:  while executing command on localhost:57640
+COPY copy_test, line 1: "0, 0"
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT * FROM copy_test;
+ key | value 
+-----+-------
+   0 |     0
+   1 |     1
+   2 |     4
+   3 |     9
+(4 rows)
+
+-- the COPY data fails
+SELECT citus.mitmproxy('conn.onCopyData().killall()');
```

Comment:
Maybe add one more step
```
SELECT citus.mitmproxy('conn.onCommandComplete(command="COPY 4").kill()');
```

Reply:
I don't see a lot of value in this test because, from the coordinator's POV, it looks exactly the same as the `conn.onCopyData().killall()` test, but I've just pushed a commit which adds it.